CMakeLists.txt

```diff
@@ -266,3 +266,19 @@ INSTALL (
     DESTINATION "${cmake_config_path}"
     COMPONENT msgpack-cxx
 )
+
+# Generate the pkgconfig file:
+SET(PKG_CONFIG_LIBS
+    "-L\${libdir}"
+)
+SET(PKG_CONFIG_CFLAGS
+    "-I\${includedir}"
```

Comment:
This is incomplete, since this misses the compilation flags. This is actually a bit tricky and why I never went further.

The obvious ones are the target compile option (`MSGPACK_USE_X3_PARSE` is set needs fixing since it is set via `CMAKE_CXX_FLAGS`), the more tricky one is that you actually also needs to handle the Boost deps here.

There are two options:
- export all the Boost flags to the .pc file (if there are any, would have to be checked)
- never enable Boost (not nice since it breaks common workflows)

I would also move the default `-I` and `-L` flags directly in the pkg-config file as well, since they are static anyway and really only clutter the CMakeLists.txt